### DISCUSSION TITLE
What may cause javascript out of memory when docker theia-ide

### DISCUSSION THREAD

**Discussion by dongysxd** [eclipse-theia/theia#14052] [Q&A]
hi,
   I create my app based on theia-ide, and I make a docker it with follow command
```
docker build -t my-app -f browser.Dockerfile .
```
I get follow error at last step
![image](https://github.com/user-attachments/assets/fac93c18-cce0-4e47-8c38-f1e264c394f5)

My computer has 32GB memory, I use follow command to check memory used to of my app in vscode
```
node -e 'console.log(v8.getHeapStatistics().heap_size_limit/(1024*1024))'
```
it return 4144, it is about 4G. I change the memory to 8G using follow command
```
$env:NODE_OPTIONS="--max-old-space-size=8196"
```
but I still get  javascript out of memory, I just add four theia extensions to my app. I think my computer memory should be enough
What may cause the out of memory. Does anyone knows?

**Comment by wss29** [comment]
add ENV node_options = "--max-old-space-size=4096" to browser.Dockerfile can solve this problem
![image](https://github.com/user-attachments/assets/2b51d645-a30e-4440-a4ea-c19dd02ddd35)
